### ISSUE TITLE
visualize does not work with 0.7.5

### ISSUE THREAD

**Issue by docastage** [emedvedev/attention-ocr#122]
Hello, 
       I have installed aocr 0.7.5 which has the visualizations.py. But I do not see visualize working for my test data. Test is working fine and I can see the accuracy. I also don't find any images getting stored in results folder. Anyone has experienced this before?Please help.

**Comment by emedvedev** [comment]
Did you use the `--visualize` flag? What's the full command you use for testing, and what's its output?

**Comment by docastage** [comment]
command:
aocr test --visualize /home/atana/FID-crop-grayscale.tfrecords --max-prediction=35 --full-ascii --use-gru --max-width=500 --max-height=100

Output:
2018-11-28 12:00:16.473767: I tensorflow/core/platform/cpu_feature_guard.cc:141] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA
2018-11-28 12:00:16.723369: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1405] Found device 0 with properties: 
name: GeForce GTX 1080 Ti major: 6 minor: 1 memoryClockRate(GHz): 1.607
pciBusID: 0000:07:00.0
totalMemory: 10.91GiB freeMemory: 10.46GiB
2018-11-28 12:00:16.723420: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1484] Adding visible gpu devices: 0
2018-11-28 12:00:17.011770: I tensorflow/core/common_runtime/gpu/gpu_device.cc:965] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-28 12:00:17.011827: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971]      0 
2018-11-28 12:00:17.011833: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] 0:   N 
2018-11-28 12:00:17.012079: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1097] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10116 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:07:00.0, compute capability: 6.1)
2018-11-28 12:00:17,118 root  INFO     phase: test
2018-11-28 12:00:17,118 root  INFO     model_dir: ./checkpoints
2018-11-28 12:00:17,119 root  INFO     load_model: True
2018-11-28 12:00:17,119 root  INFO     output_dir: ./results
2018-11-28 12:00:17,119 root  INFO     steps_per_checkpoint: 0
2018-11-28 12:00:17,119 root  INFO     batch_size: 1
2018-11-28 12:00:17,119 root  INFO     learning_rate: 1.000000
2018-11-28 12:00:17,119 root  INFO     reg_val: 0
2018-11-28 12:00:17,119 root  INFO     max_gradient_norm: 5.000000
2018-11-28 12:00:17,119 root  INFO     clip_gradients: True
2018-11-28 12:00:17,119 root  INFO     max_image_width 500.000000
2018-11-28 12:00:17,119 root  INFO     max_prediction_length 35.000000
2018-11-28 12:00:17,119 root  INFO     channels: 1
2018-11-28 12:00:17,119 root  INFO     target_embedding_size: 10.000000
2018-11-28 12:00:17,119 root  INFO     attn_num_hidden: 128
2018-11-28 12:00:17,119 root  INFO     attn_num_layers: 2
2018-11-28 12:00:17,120 root  INFO     visualize: True
2018-11-28 12:00:17,120 root  INFO     using GRU in the decoder.
using GRU CELL in decoder
2018-11-28 12:00:20,235 root  INFO     Reading model parameters from ./checkpoints/model.ckpt-190688
2018-11-28 12:00:20.561457: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1484] Adding visible gpu devices: 0
2018-11-28 12:00:20.561514: I tensorflow/core/common_runtime/gpu/gpu_device.cc:965] Device interconnect StreamExecutor with strength 1 edge matrix:
2018-11-28 12:00:20.561521: I tensorflow/core/common_runtime/gpu/gpu_device.cc:971]      0 
2018-11-28 12:00:20.561525: I tensorflow/core/common_runtime/gpu/gpu_device.cc:984] 0:   N 
2018-11-28 12:00:20.561686: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1097] Created TensorFlow device (/job:localhost/replica:0/task:0/device:GPU:0 with 10116 MB memory) -> physical GPU (device: 0, name: GeForce GTX 1080 Ti, pci bus id: 0000:07:00.0, compute capability: 6.1)
2018-11-28 12:00:22,132 root  INFO     Step 1 (1.283s). Accuracy: 100.00%, loss: 0.000000, perplexity: 1.00000, probability: 100.00% 100% (SANDRINE)
2018-11-28 12:00:22,272 root  INFO     Step 2 (0.049s). Accuracy: 100.00%, loss: 0.000003, perplexity: 1.00000, probability: 100.00% 100% (CHRISTIANE)
2018-11-28 12:00:22,508 root  INFO     Step 3 (0.047s). Accuracy: 100.00%, loss: 0.000001, perplexity: 1.00000, probability: 100.00% 100% (PASCAL)
2018-11-28 12:00:22,758 root  INFO     Step 4 (0.061s). Accuracy: 100.00%, loss: 0.000043, perplexity: 1.00004, probability: 99.96% 100% (HAUSSART)
2018-11-28 12:00:22,912 root  INFO     Step 5 (0.066s). Accuracy: 91.43%, loss: 2.037197, perplexity: 7.66909, probability: 87.82%  57% (PAULHM vs PAULINE) 
2018-11-28 12:00:23,071 root  INFO     Step 6 (0.073s). Accuracy: 92.86%, loss: 0.004201, perplexity: 1.00421, probability: 96.70% 100% (LAURENT)
2018-11-28 12:00:23,388 root  INFO     Step 7 (0.083s). Accuracy: 93.88%, loss: 0.003422, perplexity: 1.00343, probability: 97.63% 100% (PASCAL)
2018-11-28 12:00:23,529 root  INFO     Step 8 (0.051s). Accuracy: 92.86%, loss: 0.751065, perplexity: 2.11925, probability: 90.15%  86% (UILBERT vs GILBERT) 
2018-11-28 12:00:23,746 root  INFO     Step 9 (0.050s). Accuracy: 93.65%, loss: 0.001462, perplexity: 1.00146, probability: 98.98% 100% (HUGUES)
2018-11-28 12:00:23,895 root  INFO     Step 10 (0.062s). Accuracy: 92.29%, loss: 0.576144, perplexity: 1.77916, probability: 84.52%  80% (IERVE vs HERVE) 
2018-11-28 12:00:24,106 root  INFO     Step 11 (0.054s). Accuracy: 92.99%, loss: 0.011913, perplexity: 1.01198, probability: 89.83% 100% (CHRYSTEL)

**Comment by emedvedev** [comment]
The images should be stored in the `out` folder: is it completely empty? See https://github.com/emedvedev/attention-ocr/blob/master/aocr/model/model.py#L334-L344.

**Comment by docastage** [comment]
Yeah. Thank you.I can see the images as .gif files. Why have you written that "you can visualize the attention results during testing (saved to results/ by default)"? Any specific reason?

**Comment by emedvedev** [comment]
Good catch, the README just hasn't been updated together with the visualization code. Fixed it, now it correctly points to `out/`. :)
Thanks a lot!
